packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -142,25 +143,17 @@ function SelectionTool(props: Props) {
     'Escape',
     () => {
       setStartPoint(undefined);
-      setEndPoint(undefined);
+      setSelection(undefined);
     },
     [],
     { event: 'keydown' }
   );
 
-  if (!startPoint || !endPoint || !isModifierKeyPressed) {
+  if (!selection || !isModifierKeyPressed) {
     return null;
   }
 
-  const worldEndPoint = dataToWorld(endPoint);
-  const worldStartPoint = dataToWorld(startPoint);
```

Comment:
The main reason for all this (but there is more coming): we no longer compute the selection points in world space twice in a row on every mouse move. Also, we can pass the entire selection object below which saves a few lines.